docs/AddNewOp.md

```diff
@@ -35,6 +35,9 @@ ambiguity, adding the links to refs if necessary
 6. type constraints about input and output tensors
 7. shape inference function Example:
 https://github.com/onnx/onnx/blob/master/onnx/defs/nn/defs.cc#L1183
+All operators MUST implement a shape inference function before it is accepted into ONNX.
+In cases where shape inference is not possible, it must have logic to perform 
+rank inference at the very least (adding right amount of dimensions to the output shape)
```

Comment:
Can we also add that if the shape cannot be determined for whatever reasons, we shouldn’t throw an exception?

Reply:
As discussed offline - (FYI for others) - I would refrain from discouraging contributors to not throw exceptions in case of bad inputs/attribute data. For legitimate cases, exceptions should be thrown, isn't it ?

Also this seems like a detail - something that should be caught in PR reviews (should a user throw an exception here (is this bad input?) or should the user just return back gracefully with a no-op (not enough data to proceed?). Such things can be clarified at code-review time.)

Reply:
Sure.